packages/kit/src/utils/promises.js

```diff
@@ -0,0 +1,11 @@
+/**
+ * Given an object, return a new object where all top level values are awaited
+ *
+ * @param {Record<string, any>} object
+ * @returns {Promise<Record<string, any>>}
+ */
+export async function unwrap_promises(object) {
+	return Object.fromEntries(
+		await Promise.all(Object.entries(object).map(async ([key, value]) => [key, await value]))
+	);
```

Comment:
my other suggested optimisation had a subtle bug — keys would be out of order. this feels nice and simple

```suggestion
	for (const key in object) {
		if (typeof object[key]?.then === 'function') {
			return Object.fromEntries(
				await Promise.all(Object.entries(object).map(async ([key, value]) => [key, await value]))
			);
		}
	}

	return object;
```